### PR TITLE
Add sprig functions

### PR DESCRIPTION
**Pull request by kfox1111**
Fixes: https://github.com/spiffe/spire/issues/5575

### CODE REVIEW

File: pkg/common/agentpathtemplate/template.go [hunk 1]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
...
+	"mustToRawJson",
+	"ternary",
+	"deepCopy",
+	"mustDeepCopy",
+	"typeOf",
+	"typeIs",
+	"typeIsLike",
+	"kindOf",
+	"kindIs",
+	"deepEqual",
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:80]: I'd probably leave out the reflection funcs for now?

File: pkg/common/agentpathtemplate/template.go [hunk 2]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
...
 // Parse parses an agent path template. It changes the behavior for missing
 // keys to return an error instead of the default behavior, which renders a
 // value that requires percent-encoding to include in a URI, which is against
 // the SPIFFE specification.
 func Parse(text string) (*Template, error) {
-	tmpl, err := template.New("agent-path").Option("missingkey=error").Parse(text)
+	sprigMap := sprig.TxtFuncMap()
+	ourMap := make(template.FuncMap)
+	for _, f := range funcList {
+		ourMap[f] = sprigMap[f]
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:161]: We should fail if the expected function does not exist in sprigMap (e.g. we had a typo or a function was removed)

File: pkg/common/agentpathtemplate/template.go [hunk 3]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
...
+
 // Parse parses an agent path template. It changes the behavior for missing
 // keys to return an error instead of the default behavior, which renders a
 // value that requires percent-encoding to include in a URI, which is against
 // the SPIFFE specification.
 func Parse(text string) (*Template, error) {
-	tmpl, err := template.New("agent-path").Option("missingkey=error").Parse(text)
+	sprigMap := sprig.TxtFuncMap()
+	ourMap := make(template.FuncMap)
+	for _, f := range funcList {
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:160]: I'd probably construct `ourMap` once at package initialization time?

File: pkg/common/agentpathtemplate/template.go [hunk 4]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
 import (
 	"bytes"
 	"text/template"
+
+	sprig "github.com/Masterminds/sprig/v3"
 )
 
+var funcList = [...]string{
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:10]:
```suggestion
var funcList = []string{
```

File: pkg/common/agentpathtemplate/template.go [hunk 5]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
...
+	"trimSuffix",
+	"trimPrefix",
+	"nospace",
+	"initials",
+	"randAlphaNum",
+	"randAlpha",
+	"randAscii",
+	"randNumeric",
+	"swapcase",
+	"shuffle",
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:31]: 🤔 I'd probably leave this one out?

File: pkg/common/agentpathtemplate/template.go [hunk 6]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
...
+	"empty",
+	"coalesce",
+	"all",
+	"any",
+	"compact",
+	"mustCompact",
+	"fromJson",
+	"toJson",
+	"toPrettyJson",
+	"toRawJson",
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:67]: Can't imagine wanting to convert something to JSON inside of a path template...

File: pkg/common/agentpathtemplate/template.go [hunk 7]
```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
...
+	"mustFromJson",
+	"mustToJson",
+	"mustToPrettyJson",
+	"mustToRawJson",
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:71]: Can't imagine wanting to convert something to JSON inside of a path template...

File: pkg/common/agentpathtemplate/template.go [hunk 8]
```diff
@@ -150,16 +128,23 @@ var funcList = [...]string{
...
 
+var ourMap = make(template.FuncMap)
+
+func init() {
+	sprigMap := sprig.TxtFuncMap()
+	for _, f := range funcList {
+		if _, ok := sprigMap[f]; !ok {
+			panic("missing sprig function")
+		}
+		ourMap[f] = sprigMap[f]
```
Comment (azdagron) [inline pkg/common/agentpathtemplate/template.go:139]:
```suggestion
		if fn, ok := sprigMap[f]; ok {
			ourMap[f] = fn
		} else {		
			panic(fmt.Errorf("missing sprig function %q", f))
		}
		
```

**Comment by azdagron** [comment]
Hmm, looking at the failing windows integration test, I think we're falling victim to package init order....

**Comment by kfox1111** [comment]
The string its failing on (sha512sum) isn't in the codebase anywhere anymore.... It was in an earlier version of the pr. Could the vm be persisting a bad pr somehow?

**Review by azdagron — APPROVED** [review]

**Pull request merged**